File: test/ParseTreeTest.cpp
#include <gtest/gtest.h>
#include "../src/ParseTree.h"

TEST(ParseTreeTest,parseTest) {

    // Parse a tree
    ParseTree tree(
            R"(Let VB ROOT
 +-- be VB ccomp
     +-- there EX expl
     +-- cubes NNS nsubj
         +-- 15 CD num)");

    // Check whether the root word was parsed correctly
    ASSERT_EQ(tree.getRootWord().getRootWord(), "Let");

    ASSERT_EQ(tree.getRootWord().getRole(), "ROOT");

    ASSERT_EQ(tree.getRootWord().getNature(), "VB");

    // Randomly check tree structure
    ASSERT_EQ(tree.getRootWord().children.size(), 1);

    ASSERT_EQ(tree.getRootWord().children[0].children.size(), 2);

    // Check correct parsing of one of the children
    ASSERT_EQ(tree.getRootWord().children[0].children[1].getRootWord(), "cubes");

    ASSERT_EQ(tree.getRootWord().children[0].children[1].getRole(), "nsubj");

    ASSERT_EQ(tree.getRootWord().children[0].children[1].getNature(), "NNS");


}

TEST(ParseTreeTest,parseTestOnlyRoot) {

    ParseTree tree(
            R"(Let VB ROOT)");

    ASSERT_EQ(tree.getRootWord().getRootWord(), "Let");

    ASSERT_EQ(tree.getRootWord().getRole(), "ROOT");

    ASSERT_EQ(tree.getRootWord().getNature(), "VB");

    ASSERT_EQ(tree.getRootWord().children.size(), 0);

}

File: test/ParseTreeTest.java
import org.junit.Assert;
import org.junit.Test;

public class ParseTreeTest {

    @Test
    public void parseTest() {

        // Parse a tree
        ParseTree tree = new ParseTree(
                  "Let VB ROOT\n"
                + " +-- be VB ccomp\n"
                + "     +-- there EX expl\n"
                + "     +-- cubes NNS nsubj\n"
                + "         +-- 15 CD num");

        // Check whether the root word was parsed correctly
        Assert.assertEquals("Let", tree.getRootWord().getRootWord());

        Assert.assertEquals("ROOT", tree.getRootWord().getRole());

        Assert.assertEquals("VB", tree.getRootWord().getNature());

        // Randomly check tree structure
        Assert.assertEquals(1, tree.getRootWord().children.size());

        Assert.assertEquals(2, tree.getRootWord().children.get(0).children.size());

        // Check correct parsing of one of the children
        Assert.assertEquals("cubes", tree.getRootWord().children.get(0).children.get(1).getRootWord());

        Assert.assertEquals("nsubj", tree.getRootWord().children.get(0).children.get(1).getRole());

        Assert.assertEquals("NNS", tree.getRootWord().children.get(0).children.get(1).getNature());


    }

    @Test
    public void parseTestOnlyRoot() {

        ParseTree tree = new ParseTree("Let VB ROOT");

        Assert.assertEquals("Let", tree.getRootWord().getRootWord());

        Assert.assertEquals("ROOT", tree.getRootWord().getRole());

        Assert.assertEquals("VB", tree.getRootWord().getNature());

        Assert.assertEquals(0, tree.getRootWord().children.size());

    }

}